src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -177,6 +221,47 @@ public ExtractInterfaceResult ExtractInterfaceFromAnalyzedType(ExtractInterfaceT
                 navigationDocumentId: interfaceDocumentId);
         }
 
+        private async Task<ExtractInterfaceResult> ExtractInterfaceToSameFileAsync(
+            Solution solution, INamedTypeSymbol extractedInterfaceSymbol, SyntaxNode typeNode, ProjectId projectId,
+            DocumentId documentId, INamedTypeSymbol typeToExtractFrom, IEnumerable<ISymbol> includedMembers, CancellationToken cancellationToken)
+        {
+            var symbolToDeclarationAnnotationMap = CreateSymbolToDeclarationAnnotationMap(
+                includedMembers,
+                ref solution,
+                out var documentIds,
+                typeNode,
+                out var typeNodeSyntaxAnnotation,
+                cancellationToken);
+
+            var document = solution.GetDocument(documentId);
+            var originalRoot = document.GetSyntaxRootSynchronously(cancellationToken);
+            var typeDeclaration = originalRoot.GetAnnotatedNodes(typeNodeSyntaxAnnotation).Single();
+
```

Comment:
is it worth throwing?

Reply:
Not sure I understand. What would I be throwing? If no annotated nodes exist?

Reply:
you have single() at the end. so if tree contains no annotation or more than 1 annotation since node got shared more than once or deleted, than it throws. so asking whether FirstOrDefault() and handle it gracefully is better or just throwing is better.

Reply:
In this case @ryzngard controls every transformations step that happens.  So it would be an error if somehow the node he annotated didn't end up in the final tree (or was duplicated).  In that case, there is no real need for resiliency because not being 'single' would indicate a broken invariant in these subsystems.

Reply:
Correct. It is "exceptional" if we hit that.

Reply:
I get that it is exceptional. question was is that exceptional worth enough to crash VS was my question.

we used to crash VS if type colorization wasn't right. it was exceptional since it shouldn't happen, but happened due to a bug. question was whether that exceptional state was critical enough to crash VS.

my question is when that exceptional situation happens which I guess, extract interface working incorrectly. is that worth crash VS?

Reply:
tagging @jinujoseph and @ManishJayaswal since we were talking about crashing VS ever is worth ever.

Reply:
ignore the comment, if the throwing didn't crash the VS. in that case, please make sure it send out non fatal watson.

Reply:
> tagging @jinujoseph and @ManishJayaswal since we were talking about crashing VS ever is worth ever.

I don't know what this means.  Any line of code can throw an exception if any of the invariants it depends on get broken.  We don't write code as if the invariants aren't held.  It's arduous and fairly ineffective.  Instead, we write against invariants and if we run into a problem, we fix the underlying subsystem that is broken.